app/javascript/dashboard/components-next/Conversation/Chips/TypingIndicator.vue

```diff
@@ -0,0 +1,58 @@
+<script setup>
+import { useFunctionGetter, useMapGetter } from 'dashboard/composables/store';
+import { useI18n } from 'vue-i18n';
+import { computed } from 'vue';
+
+const { t } = useI18n();
+
+const getTypingUsersText = (users = []) => {
+  const count = users.length;
+  const [firstUser, secondUser] = users;
+
+  if (count === 1) {
+    return t('TYPING.ONE', { user: firstUser.name });
+  }
+
+  if (count === 2) {
+    return t('TYPING.TWO', {
+      user: firstUser.name,
+      secondUser: secondUser.name,
+    });
+  }
+
+  return t('TYPING.MULTIPLE', { user: firstUser.name, count: count - 1 });
+};
+
+const currentChat = useMapGetter('getSelectedChat');
+const typingUsersList = useFunctionGetter(
+  'conversationTypingStatus/getUserList',
+  currentChat.value.id
+);
+
+const isAnyoneTyping = computed(() => {
+  return typingUsersList.value.length !== 0;
+});
+
+const typingUserNames = computed(() => {
+  if (typingUsersList.value.length) {
+    return getTypingUsersText(typingUsersList.value);
+  }
+  return '';
+});
+</script>
+
+<template>
+  <div v-show="isAnyoneTyping">
+    <div
+      v-if="isAnyoneTyping"
```

Comment:
Redundant condition?

Reply:
Not exactly, the top level v-show is because we don't allow top-level `v-if`. And then we don't want anything to render inside, so the v-if prevents that